1.Write a line of code that will print your name.

In [39]:
print("Spam Eggsen")

Spam Eggsen


2.How do you enter a comment in a program? 

In [ ]:
# This is a comment on one line
# Now we have 2 lines of comments

In [40]:
print("Spam Eggsen") # This comment is on same line as the code

Spam Eggsen


In [ ]:
"""
This is how we can
make many
lines of comments
easily!
"""

3.What do the following lines of code output? ALSO: Why do they give a different answer? 

In [2]:
print(2 / 3) # This is division and returns a float

0.6666666666666666


In [3]:
print(2 // 3) # This modulus and since 2 divided by 3 is less than 1 it will return 0.
# It will return an int rounded down.

0


4.Write a line of code that creates a variable called pi and setsit to an appropriate value.

In [4]:
pi = 22/7
print(pi)

3.142857142857143


5.Why does this code not work? 
A = 22
print(a)

Answer: Because variable a has not been declared. A and a are not the same variable. Python is case sensitive.

6.All of the variable names below can be used. But which ONE of these isthe better variable name to use? 

a
A
Area
AREA
area
area_of_rectangle
Area_Of_Rectangle
 
Answer: area_of_rectangle because it follow the Python convention with lower cases and underscore in between words and it describes the variable better than a or area.

7.Which of these variables names are not allowed in Python? (More than one might be wrong. Also, this question is not asking about improper names, just names that aren't allowed. Test them if you aren't sure.) 

apple
Apple
APPLE
Apple2
1Apple # No number as starting a variable name
account number
account_number
account.number #No dots - its calling a function
accountNumber
account# # No comment sign at the end of the name - it will not be defined
pi
PI
fred
Fred
GreatBigVariable
greatBigVariable
great_big_variable
great.big.variable #No dots - its calling a function
2x # No number as starting a variable name
x2x
total% # No operator in the name - as only variable name can be on the left side of the expression
#left # No comment sign at the beginning of the name - nothing will be happen
 


8.Why does this code not work? 

print(a)
a = 45
 
Answer: a has to be defined before it can be printed. Reverse the 2 lines and it will work.

9.Explain the mistake in this code: 

pi = float(3.14)
 
Answer: The float function is not necessary - 3.14 is already a float. It is enough to write:
pi = 3.14

10.This program runs, but the code still could be better. Explain what is wrong with the code. 

radius = float(input("Radius:"))
x = 3.14
pi = x
area = pi  * radius ** 2
print(area)
 
Answer:
The code can be one line shorter with 
pi = 3.14
Also it would help with more comments to understand the code

11.Explain the mistake in the following code: 

x = 4
y = 5
a = ((x) * (y))
print(a)
 
Answer: The paranthesises are not necessary, it is enough to write:
a = x * y

12.Explain the mistake in the following code: 

x = 4
y = 5
a = 3(x + y)
print(a)
 
Answer: Python cant be written like math with 3x but has to be written as 3 * x. The correct code is:
a = 3 * (x + y)

13.Explain the mistake in the following code: 

radius = input(float("Enter the radius:"))

In [23]:
# Answer to 13: the float and input functions has to change places.
radius = float(input("Enter the radius:"))

Enter the radius:20


14.Do all these print the same value? Which one is better to use and why? 

print(2/3+4)
print(2 / 3 + 4)
print(   2 /    3+    4  )
 
Answer: Yes the number of spaces or lack of space does not affect the evaluation. It is best to use the second line, because it has the highest readability.

15.What is a constant? 

Answer: A constant is a variable with a value that will never change - and could be written like this:
SPEED_OF_LIGHT = 300000 # kilometer/second

16.How are variable names for constants different than other variable names? 

Answer: They are written in all upper case letters.

17.What is a single quote and what is a double quote? Give and label an example of both. 

In [41]:
# Answer to 17:
# Example of single quote:
name = 'Spam'
# Example of double quote:
name = "Spam"
# Both can be used interchangeably - and they can be used to print each other:
string = "His name was 'Spam'"
print(string)

His name was 'Spam'


18.Write a Python program that will use escape codes to print a double-quote and a new line using the Window's standard. (Note: I'm asking for the Window's standard here. Look it up out of Chapter 1.)

In [42]:
# Answer to 18:
print("His first name was \"Spam\" and \n his last name was \"Eggsen\"")

His first name was "Spam" and 
 his last name was "Eggsen"


19.Can a Python program print text to the screen using single quotes instead of double quotes? 

In [27]:
# Answer to 19: Yes
print('Printing using single quotes')

Printing using single quotes


20.Why does this code not calculate the average? 

print(3 + 4 + 5 / 3)

Answer: Python will do division before addition - so it will evaluate 5 / 3 and then add 3 + 4 + ( 5 / 3)

In [28]:
# Answer to 20: The correct code is:
print((3 + 4 + 5) / 3)

4.0


21.What is an ``operator'' in Python? 

Answer: An operator is a symbol like + or - or * or / or // (floor division) or % (modulus) or ** (power).
Operators are special symbols in Python that carry out arithmetic or logical computation. The value that the operator operates on is called the operand

22.What does the following program print out? 

x = 3
x + 1
print(x)

Answer: 3, because there is no assignment operator (fx =) in line 2. The + 1 is not stored in x.

In [29]:
x = 3
x + 1
print(x)

3


23.Correct the following code: 

user_name = input("Enter your name: )"

In [43]:
# Answer to 23: double quote and end paranthesis change places:
user_name = input("Enter your name: ")

Enter your name: Spam


24.Correct the following code: 

value = int(input(print("Enter your age")))

In [44]:
# Answer to 24: Remove the print function - else variable value will not be assigned a value

value = int(input("Enter your age "))

Enter your age 100
